### PR TITLE
Optimize torch.ops.ns.opname.overload accessor in torch dispatch

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85133
* __->__ #85132

This doesn't actually seem to help all that much.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: c10/core/PyHandleCache.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      // attempt to claim this cache entry with the specified interpreter tag
+      if (pyinterpreter_.compare_exchange_strong(
+              expected, self_interpreter, std::memory_order_acq_rel)) {
+        data_ = r;
+      }
+      // This shouldn't be possible, as you should be GIL protected
+      TORCH_INTERNAL_ASSERT(expected != self_interpreter);
+      return r;
+    } else {
+      return slow_accessor();
```
Comment (wconstab) [inline c10/core/PyHandleCache.h:66]: we're not actually making torchdeploy any slower, we're just speeding up single-interpreter use cases and leaving torchdeploy behind.  right?

File: torch/csrc/autograd/python_variable.cpp [hunk 2]
```diff
@@ -2263,9 +2267,9 @@ void ConcretePyInterpreterVTable::dispatch(
       overloaded_args,
       args.ptr(),
       kwargs.ptr(),
-      func_name,
+      nullptr,
```
Comment (wconstab) [inline torch/csrc/autograd/python_variable.cpp:2270]: hmm...

**Review by wconstab — APPROVED** [review]
nice.  how much does this help btw?

File: c10/core/PyHandleCache.h (hunk 1 above)
Comment (ezyang) [reply c10/core/PyHandleCache.h:66]: yep

**Comment by ezyang** [comment]
It did not perceptibly help in the microbenchmark lol

**Pull request closed without merging**